### PR TITLE
Add log rotation for http logs.

### PR DESCRIPTION
**Pull request by GDLMadushanka**
Fixes wso2/product-ei/issues/2420

Default value is false

### CODE REVIEW

**Comment by milindaperera** [comment]
We need to update documentation as well. Please create documentation jira as well

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java [hunk 1]
```diff
@@ -190,19 +202,50 @@ protected synchronized void open() {
...
             // If no rotate - no need for dateStamp in fileName
             if (isRotatable) {
-                pathname = dir.getAbsolutePath() + File.separator + AccessConstants.getPrefix() +
-                           dateStamp + AccessConstants.getSuffix();
-            } else {
-                pathname = dir.getAbsolutePath() + File.separator + AccessConstants.getPrefix() +
-                           AccessConstants.getSuffix();
+                File existing = new File(pathname);
+                // renaming existing file to a file with date
+                if (existing != null && existing.exists()) {
```
Comment (sdkottegoda) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: existing will never be null, right?

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java [hunk 2]
```diff
@@ -211,6 +254,29 @@ protected synchronized void open() {
         }
     }
 
+    /*
```
Comment (sdkottegoda) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: Incorrect java doc style, shall we correct?

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java [hunk 3]
```diff
@@ -190,19 +202,50 @@ protected synchronized void open() {
 
         // Open the current log file
         try {
-            String pathname;
+            Date today = new Date(System.currentTimeMillis());
+            String pathname = dir.getAbsolutePath() + File.separator + AccessConstants.getPrefix() +
```
Comment (sdkottegoda) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: Line wrapping needs to happen before an operator, shall we correct in all places?

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java [hunk 4]
```diff
@@ -211,6 +254,29 @@ protected synchronized void open() {
         }
     }
 
+    /*
+        When some error happens, create a file with yesterday timestamp
+     */
+    private void getInErrorFileName(File dir, File existing) {
```
Comment (sdkottegoda) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: the method name is not descriptive of what it does, shall we correct it?

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java [hunk 5]
```diff
@@ -211,6 +254,29 @@ protected synchronized void open() {
...
+        Date today = new Date(System.currentTimeMillis());
+        Calendar calender = Calendar.getInstance();
+        calender.setTime(today);
+        calender.add(Calendar.DATE, -1);
+        Date dayBefore = calender.getTime();
+        String pathName = dir.getAbsolutePath() + File.separator + AccessConstants.getPrefix() +
+                fileDateFormatter.format(dayBefore) + AccessConstants.getSuffix();
+        File oldFile = new File(pathName);
+        // file with yesterday timestamp exists
+        if (oldFile != null && oldFile.exists()) {
```
Comment (sdkottegoda) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: can old file be null?

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java (hunk 1 above)
Comment (GDLMadushanka) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: can be, when the server start for the first time

Comment (GDLMadushanka) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: +1 filed

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java (hunk 2 above)
Comment (GDLMadushanka) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: +1 filed

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java (hunk 3 above)
Comment (GDLMadushanka) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: +1 filed

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java (hunk 4 above)
Comment (GDLMadushanka) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: +1 filed

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java (hunk 5 above)
Comment (GDLMadushanka) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/access/AccessLogger.java]: +1 filed

**Pull request merged**